packages/web-components/fast-foundation/src/listbox/listbox.ts

```diff
@@ -206,13 +228,94 @@ export abstract class Listbox extends FoundationElement {
         this.shouldSkipFocus = false;
     }
 
+    /**
+     * Returns the options which match the current typeahead buffer.
+     *
+     * @internal
+     */
+    protected getTypeaheadMatches(): ListboxOption[] {
+        const pattern = this.typeaheadBuffer.replace(/[.*+\-?^${}()|[\]\\]/g, "\\$&");
+        const re = new RegExp(`^${pattern}`, "gi");
+        return this.options.filter((o: ListboxOption) => o.text.trim().match(re));
+    }
+
+    /**
+     * Determines the index of the next option which is selectable, if any.
+     *
+     * @param prev - the previous selected index
+     * @param next - the next index to select
+     *
+     * @internal
+     */
+    protected getSelectableIndex(prev: number = this.selectedIndex, next: number) {
+        const direction = prev > next ? -1 : prev < next ? 1 : 0;
+        const potentialDirection = prev + direction;
+
+        let nextSelectableOption: ListboxOption | null = null;
+
+        switch (direction) {
+            case -1: {
+                nextSelectableOption = this.options.reduceRight(
+                    (nextSelectableOption, thisOption, index) =>
+                        !nextSelectableOption &&
+                        !thisOption.disabled &&
+                        index < potentialDirection
+                            ? thisOption
+                            : nextSelectableOption,
+                    nextSelectableOption
+                );
+                break;
+            }
+
+            case 1: {
+                nextSelectableOption = this.options.reduce(
+                    (nextSelectableOption, thisOption, index) =>
+                        !nextSelectableOption &&
+                        !thisOption.disabled &&
+                        index > potentialDirection
+                            ? thisOption
+                            : nextSelectableOption,
+                    nextSelectableOption
+                );
+                break;
+            }
+
+            default:
+            // impossible!
```

Comment:
If `0` is impossible do we need to be setting it on line 251 or can we simplify that ternary?